report_thesis/src/sections/proposed_approach/optimization_framework.tex

```diff
@@ -0,0 +1,193 @@
+Additionally, the TPE sampler allows us to control how many of the trials to be reserved for exploration, which is beneficial when the search space is large. \cite{optuna_2019}
+
+The pruner is responsible for stopping unpromising trials early, to reduce the computational costs of the optimization process and is initialized in line~\ref{step:initialize_pruner}.
+Similar to the sampler, there exist several options for the choice of pruner.
+In our case we chose to use the Hyperband pruner, due to its balanced approach when pruning.
+Initially, the pruner will consider more configurations, but allow them to run for a shorter time.
+Through several iterations, it will gradually reduce the number of configurations being considered, but allow them to run for a longer time.
+This provides a higher likelihood that promising configurations are not pruned too early, while still reducing the computational costs.\cite{optuna_2019}
+
+Guiding both the sampler and the pruner is the objective function, which evaluates the performance of each trial, seeking to either minimize or maximize some specified metrics.
+In our case we are seeking to minimize the \texttt{rmse\_cv} and \texttt{std\_dev\_cv}, defined previously.
+This can be seen in line~\ref{step:initialize_objective}.
+
+The role of the objective function is to provide the metric data that the sampler and pruner need to make informed decisions.
+As we step through each oxide and model in Lines~\ref{step:oxide_loop} to~\ref{step:model_loop}, we call the $objective$ function with the $Optimize\_Combined\_Objective$ function, seen in Line~\ref{step:optimize_combined_objective}.
+The $Optimize\_Combined\_Objective$ is returning the metrics for each trial, which the objective function then uses to guide the optimization process.
+
+This leads us to the \nameref{alg:combined_objective} function which, as previously mentioned, contains the optimization logic.
+It defines the systematic process of hyperparameter tuning, model training, and evaluation across multiple trials.
+
+To this function we supply the current model $M$, the number of trials to run $N$, the sampler, and the pruner, seen in Line~\ref{step:combined_objective_params}.
+The number of trials determines how many times the optimization process will be run for each model and oxide.
+This is an important parameter, as it regulates how much time the framework has to optimize a given model.
+In an ideal scenario, this number would be very high to ensure that the optimization process has identified the best possible configuration.
+However, depending on the number of models in consideration the number of trials can quickly become computationally prohibitive.
+
+In Lines~\ref{step:sample_hyperparameters} to~\ref{step:instantiate_dim_reduction} we instantiate the model and the preprocessors with the hyperparameters sampled by the sampler.
+The sampling process of these is what is being guided by the objective function based on the metrics returned.
```

Comment:
I think this is clearer, but you'd need to specify that it does random sampling (i think) if no trial has been run yet.

```suggestion
The sampling process is being guided by the objective function, via the returned metrics from the previous trial.
```